cognite/neat/rules/_constants.py

```diff
@@ -0,0 +1,173 @@
+import re
+import sys
+from functools import cached_property
+
+if sys.version_info >= (3, 11):
+    from enum import StrEnum
+else:
+    from backports.strenum import StrEnum
+
+
+class EntityTypes(StrEnum):
+    view_non_versioned = "view_non_versioned"
+    subject = "subject"
+    predicate = "predicate"
+    object = "object"
+    class_ = "class"
+    parent_class = "parent_class"
+    property_ = "property"
+    dms_property = "dms_property"
+    information_property = "information_property"
+    object_property = "ObjectProperty"
+    data_property = "DatatypeProperty"
+    annotation_property = "AnnotationProperty"
+    object_value_type = "object_value_type"
+    data_value_type = "data_value_type"  # these are strings, floats, ...
+    xsd_value_type = "xsd_value_type"
+    dms_value_type = "dms_value_type"
+    dms_node = "dms_node"
+    view = "view"
+    reference_entity = "reference_entity"
+    container = "container"
+    datamodel = "datamodel"
+    undefined = "undefined"
+    multi_value_type = "multi_value_type"
+    asset = "asset"
+    relationship = "relationship"
+    edge = "edge"
+    reverse = "reverse"
+    unit = "unit"
+    version = "version"
+    prefix = "prefix"
+
+
+def get_reserved_words() -> dict[str, list[str]]:
+    return {
+        "class": ["Class", "class"],
+        "view": [
+            "Query",
+            "Mutation",
+            "Subscription",
+            "String",
+            "Int32",
+            "Int64",
+            "Int",
+            "Float32",
+            "Float64",
+            "Float",
+            "Timestamp",
+            "JSONObject",
+            "Date",
+            "Numeric",
+            "Boolean",
+            "PageInfo",
+            "File",
+            "Sequence",
+            "TimeSeries",
+        ],
+        "property": [
+            "property",
+            "space",
+            "externalId",
+            "createdTime",
+            "lastUpdatedTime",
+            "deletedTime",
+            "edge_id",
+            "node_id",
+            "project_id",
+            "property_group",
+            "seq",
+            "tg_table_name",
+            "extensions",
+        ],
+    }
+
+
+ENTITY_PATTERN = re.compile(r"^(?P<prefix>.*?):?(?P<suffix>[^(:]*)(\((?P<content>.+)\))?$")
```

Comment:
I am not against it, more curious.

What is the rational for moving all the regex away from utils and into _constants?